public_admin/templates/admin/base.html

```diff
@@ -1,4 +1,4 @@
 {% extends "admin/base.html" %}
 
-{% block welcome-msg %}{% endblock %}
-{% block userlinks %}{% endblock %}
+{% block welcome-msg %}% if not request.user.is_anonymous %}{{ block.super }}{% endif %}{% endblock %}
```

Comment:
I believe that inside the `welcome-msg` block, and inside the `if` you added, we have to add [the code from the original admin template](https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/base.html#L38-L41), don't we?

Reply:
`{{ block.super }}` does that, isn't it?

Reply:
Ah… true story. I was doing something stupid which was reading code on the mobile, and missed that. 100% my bad. Sorry, mate!